eng/pipelines/templates/jobs/vmr-build.yml

```
@@ -602,10 +617,18 @@ jobs:
         testRunTitle: ScenarioTests_$(Agent.JobName)
 
   - ${{ if or(ne(variables['System.TeamProject'], 'internal'), eq(variables['Build.Reason'], 'PullRequest')) }}:
+    # Both publishing steps are necessary to ensure artifacts are published on both success and failure.
+    # This prevents overwrite conflicts in the event of a failed build followed by a rerun and a successful build.
```

Successful or not, the conflict can occur.

```suggestion
    # This prevents overwrite conflicts in the event of a failed build followed by a rerun.
```